tools/build-plugin.js

```diff
@@ -202,16 +201,6 @@ export class SourceProcessorSet {
     // don't use iteration functions, so we can return (and start at #1)
     for (let i = 1; i < parts.length; i++) {
       const extension = parts.slice(i).join('.');
-      // We specially handle 'js' in the build tool, because you can't provide a
-      // plugin to handle 'js' files, because the plugin would need to be built
-      // with JavaScript itself!  Places that hardcode JS are tagged with
-      // #HardcodeJs.
```

Comment:
Hmm, I think removing `this._hardcodeJs` isn't correct, although I think maybe it was a bad name for the option to this class. The point here is that there is a default compiler for `*.js` but no default linter (unless you changed that in this PR).  By dropping `this._hardcodeJs` it will now make the linter code think that there is a linter for `*.js` even if there isn't, which isn't correct (and probably will cause it to crash when it looks for a sourceProcessors field on the returned classification).

(Not doing a full review this minute, was just poking to make sure you were working on a suitably new version of code (w/ SourceProcessorSet) and you are.)